src/include/cephfs/types.h

```diff
@@ -547,6 +553,8 @@ struct inode_t {
 
   bufferlist fscrypt_last_block;
 
+  inodeno_t remote_ino = 0; // referent inode - remote inode link
```

Comment:
Do I understand correctly that `remote_ino` and `referent_inodes` aren't ever going to be used simultaneously? If so, then these fields should be under a union to not waste space.
Alternatively, the remote_ino could be a single member of the collection if the inode represents a referent. The collection could be called `remote_inodes` and treated respectively to the inode type.